packages/openstef-models/src/openstef_models/feature_engineering/weather_transforms/air_related_features_transform.py

```diff
@@ -0,0 +1,196 @@
+# SPDX-FileCopyrightText: 2025 Contributors to the OpenSTEF project <short.term.energy.forecasts@alliander.com>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+"""Transform for calculating and adding air-related meteorological features to a time series dataset.
+
+The transform computes saturation vapour pressure, vapour pressure, dewpoint, and air density
+based on temperature, pressure, and relative humidity columns using established physical equations.
+"""
+
+from typing import Any, Literal, override
+
+import numpy as np
+import pandas as pd
+from pydantic import Field
+
+from openstef_core.base_model import BaseConfig
+from openstef_core.datasets import TimeSeriesDataset
+from openstef_core.datasets.transforms import TimeSeriesTransform
+from openstef_core.exceptions import MissingColumnsError
+
+type AirRelatedFeatureName = Literal["saturation_vapour_pressure", "vapour_pressure", "dewpoint", "air_density"]
+
+
+class AirRelatedFeaturesTransform(BaseConfig, TimeSeriesTransform):
+    """Transform that calculates air-related meteorological features from basic weather data.
+
+    This transform calculates various air-related features including saturation vapour pressure,
+    vapour pressure, dewpoint, and air density using standard meteorological formulas. It requires
+    temperature, pressure, and relative humidity as input columns.
+    The calculated features can be used to enhance weather-based prediction models by providing
+    additional atmospheric state information that may correlate with energy consumption patterns.
+
+    Example:
+        >>> import pandas as pd
+        >>> from openstef_core.datasets.timeseries_dataset import TimeSeriesDataset
+        >>> from openstef_models.feature_engineering.weather_transforms.air_related_features_transform import (
+        ...     AirRelatedFeaturesTransform
+        ... )
+        >>>
+        >>> # Create sample weather data
+        >>> data = pd.DataFrame({
+        ...     'temperature': [20.0, 25.0, 15.0],
+        ...     'pressure': [1013.25, 1015.0, 1010.0],
+        ...     'relative_humidity': [60.0, 70.0, 80.0]
+        ... },
+        ... index=pd.date_range('2025-06-01 12:00:00', periods=3, freq='h'))
+        >>> dataset = TimeSeriesDataset(data=data, sample_interval=pd.Timedelta(hours=1))
+        >>>
+        >>> # Initialize transform with specific features
+        >>> transform = AirRelatedFeaturesTransform(
+        ...     included_features=["dewpoint", "air_density"]
+        ... )
+        >>>
+        >>> # Apply transformation
+        >>> result = transform.transform(dataset)
+        >>> "dewpoint" in result.data.columns
+        True
+        >>> "air_density" in result.data.columns
+        True
```

Comment:
```suggestion
        >>> "dewpoint" in result.feature_names
        True
        >>> "air_density" in result.feature_names
        True
```

We can use a more direct attribute here. Or test for all the columns.

```
>>> result.feature_names
['temperature', 'pressure', 'relative_humidity', 'dewpoint', 'air_density'] 
```